rest-service/manager_rest/rest/resources_v3_1/execution_schedules.py

```diff
@@ -0,0 +1,275 @@
+)
+from manager_rest.utils import get_formatted_timestamp
+from manager_rest.security import SecuredResource
+from manager_rest.security.authorization import authorize
+from manager_rest.resource_manager import get_resource_manager
+from manager_rest.storage import get_storage_manager, models
+
+
+class ExecutionSchedules(SecuredResource):
+    @swagger.operation(
+        responseClass='List[{0}]'.format(models.ExecutionSchedule.__name__),
+        nickname='list',
+        notes='Returns a list of existing execution schedules.'
+    )
+    @authorize('snapshot_list', allow_all_tenants=True)
+    @rest_decorators.marshal_with(models.ExecutionSchedule)
+    @rest_decorators.create_filters(models.ExecutionSchedule)
+    @rest_decorators.paginate
+    @rest_decorators.sortable(models.ExecutionSchedule)
+    @rest_decorators.all_tenants
+    @rest_decorators.search('id')
+    def get(self, _include=None, filters=None, pagination=None,
+            sort=None, all_tenants=None, search=None, **kwargs):
+        return get_storage_manager().list(
+            models.ExecutionSchedule,
+            include=_include,
+            filters=filters,
+            substr_filters=search,
+            pagination=pagination,
+            sort=sort,
+            all_tenants=all_tenants
+        )
+
+
+class ExecutionSchedulesId(SecuredResource):
+    @rest_decorators.marshal_with(models.ExecutionSchedule)
+    @authorize('execution_start')
+    def put(self, schedule_id, **kwargs):
+        """Schedule a workflow execution"""
+
+        validate_inputs({'schedule_id': schedule_id})
+        request_dict = get_json_and_verify_params({'deployment_id',
+                                                   'workflow_id',
+                                                   'since',
+                                                   'slip'})
+
+        # --- Execution parameters. Recorded as-is in the schedule event
+        workflow_id = request_dict['workflow_id']
+        execution_arguments = {
+            'allow_custom_parameters': verify_and_convert_bool(
+                'allow_custom_parameters',
+                request_dict.get('allow_custom_parameters', False)),
+            'force': verify_and_convert_bool(
+                'force',
+                request_dict.get('force', False)),
+            'dry_run': verify_and_convert_bool(
+                'dry_run',
+                request_dict.get('dry_run', False)),
+            'queue': verify_and_convert_bool(
+                'queue',
+                request_dict.get('queue', False)),
+            'wait_after_fail': request_dict.get('wait_after_fail', 600)
+        }
+        parameters = request_dict.get('parameters', None)
+        if parameters is not None and parameters.__class__ is not dict:
+            raise manager_exceptions.BadParametersError(
+                "request body's 'parameters' field must be a dict but"
+                " is of type {0}".format(parameters.__class__.__name__))
```

Comment:
```suggestion
                "parameters: expected dict, but got: {0}".format(parameters))
```
look how much shorter and without the `__` attrs! :P